ipfs-pinning-service.yaml

```diff
@@ -141,66 +193,134 @@ paths:
           type: string
+        status:
+          $ref: '#/components/schemas/Status'
+        pin:
+          $ref: '#/components/schemas/Pin'
 
-    Cid:
+    Pin:
+      description: Pin object
       type: object
       properties:
-        /:
+        cid:
+          description: CID to be pinned
           type: string
+        depth:
+          description: Defines how deep the DAG should be pinned (-1 recursive, 0 direct)
+          type: integer
+          format: int32
+          default: -1
+          minimum: -1
+        meta:
+          $ref: '#/components/schemas/Meta'
+      required:
+      - cid
+
+    Status:
+      description: Status in which pin object can exist at a pinning service
+      type: string
+      enum:
+        - pinned
+        - pinning  # in-progress
+        - removing # in-progress
+        - expired  # still pinned for some time, but run out of funds and is not provided to the IPFS network
 
-    Metadata:
+    Meta:
+      description: Optional metadata for Pin object
       type: object
       additionalProperties:
         type: string
 
-    ReplicationFactor:
-      type: integer
-      minimum: -1
-
-    PinStatus:
+    Error:
+      description: Base Error object
       type: object
       properties:
-        peername:
-          type: string
-        peer:
-          type: string
-        error:
+        code:
+          type: integer
+        message:
           type: string
-        pin:
-          $ref: '#/components/schemas/Pin'
-        status:
+
+  parameters:
+
+    skip:
+      description: number of items to skip
+      name: skip
+      in: query
+      required: false
+      schema:
+        type: integer
+        format: int32
+
+    limit:
+      description: max records to return
+      name: limit
+      in: query
+      required: false
+      schema:
+        type: integer
+        format: int32
+
+    cid:
+      description: return pin objects for the specified CID(s)
+      name: cid
+      in: query
+      required: false
+      schema:
+        type: array
+        items:
           type: string
+      style: form # ?cid=Qm1,Qm2,bafy3
+      explode: false
+      examples:
+        oneId:
+          summary: Example of a single CID
+          value: [QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR]   # ?cid=Qm
+        multipleIds:
+          summary: Example of multiple CIDs
+          value: [QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR,bafkreigtdgsgv2f3bkhsmxvku3bpnnqzubcxeupf7fff5f7l7tlm2v237a]   # ?cid=Qm,bafy
 
-    Pin:
-      type: object
-      properties:
-        cid:
-          $ref: '#/components/schemas/Cid'
-        metadata:
-          $ref: '#/components/schemas/Metadata'
-        replication:
-          $ref: '#/components/schemas/ReplicationFactor'
+    depth:
+      description: return pin objects only for pins with the specified depth
+      name: depth
+      in: query
+      required: false
```

Comment:
What kind of pin depths are returned if this is not specified? Recursive only?  Everything?